### PR TITLE
Additional CodeQL Fixes

### PR DESCRIPTION
**Pull request by TaylorBeebe**
## Description

Various fixes

- [x] Impacts functionality?
- **Functionality** - Does the change ultimately impact how firmware
functions?
- Examples: Add a new library, publish a new PPI, update an algorithm,
...
- [x] Impacts security?
- **Security** - Does the change have a direct security impact on an
application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
- **Breaking change** - Will anyone consuming this change experience a
break
    in build or boot behavior?
- Examples: Add a new library class, move a module to a different repo,
call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
- **Documentation** - Does the change contain explicit documentation
additions
    outside direct code modifications (and comments)?
- Examples: Update readme file, add feature readme file, link to
documentation
    on an a separate Web page, ...

## How This Was Tested

Build and boot changes on QemuQ35Pkg to EFI shell.

## Integration Instructions

N/A

### CODE REVIEW

**Review by makubacki — APPROVED** [review]

**Review by apop5 — APPROVED** [review]

File: ShellPkg/Application/Shell/ShellParametersProtocol.c [hunk 1]
```diff
@@ -1182,7 +1183,13 @@ UpdateStdInStdOutStdErr (
 
         if (!ErrUnicode && !EFI_ERROR (Status)) {
           TempHandle = CreateFileInterfaceFile (TempHandle, FALSE);
-          ASSERT (TempHandle != NULL);
+          // MU_CHANGE [BEGIN] - CodeQL change
+          if (TempHandle == NULL) {
+            ASSERT (TempHandle != NULL);
+            Status = EFI_INVALID_PARAMETER;
```
Comment (kenlautner) [inline ShellPkg/Application/Shell/ShellParametersProtocol.c]: What's the reason for this being EFI_INVALID_PARAMETER vs EFI_OUT_OF_RESOURCES used on line 1242?

**Review by kenlautner — APPROVED** [review]

File: ShellPkg/Application/Shell/ShellParametersProtocol.c (hunk 1 above)
Comment (TaylorBeebe) [reply ShellPkg/Application/Shell/ShellParametersProtocol.c]: Should also be EFI_OUT_OF_RESOURCES - will update thanks :)

File: NetworkPkg/Ip6Dxe/Ip6Nd.c [hunk 2]
```diff
@@ -2517,8 +2525,16 @@ Ip6ProcessRedirect (
   //
   OptionLen = (UINT16)(Head->PayloadLength - IP6_REDITECT_LENGTH);
   if (OptionLen != 0) {
+    // MU_CHANGE [BEGIN] - CodeQL change
```
Comment (spbrogan) [inline NetworkPkg/Ip6Dxe/Ip6Nd.c:2527]: what am i missing here?  this looks like a useless add

Comment (TaylorBeebe) [reply NetworkPkg/Ip6Dxe/Ip6Nd.c:2527]: Discussed offline and pushed a fix, resolving this comment

**Pull request merged**